File: python/37b_1722B_Colourblindness.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = input().strip()
        b = input().strip()
        flag = 0
        for i in range(n):
            if (a[i] == "R" and b [i] != "R") or (b[i] == "R" and a[i] != "R"):
                flag = 1
                break
        if flag == 0:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/37b_1722B_Colourblindness.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, flag;
    std::string a, b;
    std::vector<std::string> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::cin>>a>>b;
        flag = 0;
        for(int j = 0; j < n; j++){
            if ((a[j] == 'R' && b[j] != 'R') || (a[j] != 'R' && b[j] == 'R')){
                flag = 1;
                break;
            }
        }
        if (flag == 0){
            ans.push_back("YES");
        }
        else{
            ans.push_back("NO");
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}